metpy/calc/tests/test_kinematics.py

```diff
@@ -559,3 +559,66 @@ def test_convergence_vorticity():
     true_v = np.ones_like(u) / units.sec
     assert_array_equal(c, true_c)
     assert_array_equal(v, true_v)
+
+
+def test_lat_lon_grid_deltas_1d():
+    """Test for lat_lon_grid_spacing for variable grid."""
+    lat = np.arange(40, 50, 2.5)
+    lon = np.arange(-100, -90, 2.5)
+    dx, dy = lat_lon_grid_deltas(lon, lat)
+    dx_truth = np.array([[212943.5585, 212943.5585, 212943.5585],
+                         [204946.2305, 204946.2305, 204946.2305],
+                         [196558.8269, 196558.8269, 196558.8269],
+                         [187797.3216, 187797.3216, 187797.3216]]) * units.meter
+    dy_truth = np.array([[277987.1857, 277987.1857, 277987.1857, 277987.1857],
+                         [277987.1857, 277987.1857, 277987.1857, 277987.1857],
+                         [277987.1857, 277987.1857, 277987.1857, 277987.1857]]) * units.meter
+    assert_almost_equal(dx, dx_truth, 4)
+    assert_almost_equal(dy, dy_truth, 4)
+
+
+def test_lat_lon_grid_deltas_2d():
```

Comment:
I wonder if we could just parameterize on a `reversed` argument. If true, flip the arrays and flip the sign on the truth value. I mean that's all you're doing anyway.